src/test/app/CrossingLimits_test.cpp

```diff
@@ -256,6 +301,251 @@ class CrossingLimits_test : public beast::unit_test::suite
+            //     offers unfunded. b. Carol's remaining 800 offers are consumed
+            //     as unfundedleaving. c. 199 of alice's XRP(1) to USD(3) offers
+            //     are consumed.
+            //        A book step is allowed to consume a maxium of 1000 offers
+            //        at a given quality, and that limit is now reached.
+            //     d. Now the strand is dry, even though there are still funded
+            //     XRP(1) to USD(3) offers available. Bob has spent 400 EUR and
+            //     received 600 USD in this step. (200 funded offers consumed
+            //     800 unfunded offers)
+            //  4. The best is the non-autobridged offers that take 499 EUR and
+            //  give 499 USD.
+            //     Bob has 2000 EUR, and has spent 1+600+400=1001 EUR. He has
+            //     999 left. Bob spent 499 EUR and receives 499 USD.
+            // In total: Bob spent EUR(1 + 600 + 400 + 499) = EUR(1500). He
+            // started with 2000 so has 500 remaining
+            //           Bob received USD(10 + 1200 + 600 + 499) = USD(2309).
+            //           Alice spent 10 + 300*4 + 199*3 + 499 = 2306 USD. She
+            //           started with 4000 so has 1704 USD remaining. Alice
+            //           received 600 + 400 + 500 = 1500 EUR
+            env.trust(EUR(10000), bob);
+            env.close();
+            env(pay(gw, bob, EUR(2000)));
+            env.close();
+            env(offer(bob, USD(4000), EUR(4000)));
+            env.close();
+
+            env.require(balance(bob, USD(2309)));
+            env.require(balance(bob, EUR(500)));
+            env.require(offers(bob, 1));
+            env.require(owners(bob, 3));
+
+            env.require(balance(alice, USD(1694)));
+            env.require(balance(alice, EUR(1500)));
+            auto const numAOffers =
+                1 + 2000 + 300 + 1000 + 1 - (1 + 2 * 300 + 2 * 199 + 1 + 1);
+            env.require(offers(alice, numAOffers));
+            env.require(owners(alice, numAOffers + 2));
+
+            env.require(offers(carol, 0));
+        }
+    }
+
+    void testAutoBridgedLimits (FeatureBitset features)
+    {
+        // Taker and FlowCross are too different in the way they handle
+        // autobridging to make one test suit both approaches.
+        //
+        //  o Taker alternates between books, completing one full increment
+        //    before returning to make another pass.
+        //
+        //  o FlowCross extracts as much as possible in one book at one Quality
+        //    before proceeding to the other book.  This reduces the number of
+        //    times we change books.
+        //
+        // So the tests for the two forms of autobridging are separate.
+        if (features[featureFlowCross])
+            testAutoBridgedLimitsFlowCross (features);
+        else
+            testAutoBridgedLimitsTaker (features);
+    }
+
+    void
+    testOfferOverflow (FeatureBitset features)
```

Comment:
I'd like to see the Overflow test modified so it actually exercises the intended case.  That will mean that the FlowCross case will generate `tecOVERSIZE` rather than `tecSUCCESS`.  But we should be honest with ourselves.

An example of how the test could be changed is shown here: https://github.com/scottschurr/rippled/commit/8dfb7949aad455679d24d02f22ca1bfc1eac1fbe.

Reply:
Pushed a commit with a test for tecOVERSIZE